File: app/src/main/java/com/example/mike/mp3player/client/views/viewholders/MediaItemViewHolder.java
package com.example.mike.mp3player.client.views.viewholders;

import android.support.v4.media.MediaBrowserCompat.MediaItem;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.mike.mp3player.client.AlbumArtPainter;

public abstract class MediaItemViewHolder extends RecyclerView.ViewHolder {

    protected final AlbumArtPainter albumArtPainter;

    public MediaItemViewHolder(@NonNull View itemView, AlbumArtPainter albumArtPainter) {
        super(itemView);
        this.albumArtPainter = albumArtPainter;
    }

    public abstract void bindMediaItem(MediaItem item);
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/views/viewholders/MediaItemViewHolder.kt
package com.github.goldy1992.mp3player.client.views.viewholders

import android.content.Context
import android.support.v4.media.MediaBrowserCompat
import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding

abstract class MediaItemViewHolder<out T : ViewBinding>
    constructor(val context : Context,
                val binding : T)
    : RecyclerView.ViewHolder(binding.root) {

    /**
     * Binds the [MediaBrowserCompat.MediaItem] to the [binding].
     */
    abstract fun bindMediaItem(item: MediaBrowserCompat.MediaItem)

}